src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -91,32 +91,47 @@ TryDropMarkedShards(bool waitForCleanupLock)
  * will be removed at a later time when there are no locks held anymore on
  * those placements.
  *
+ * If waitForLocks is false, then if we cannot take a lock on pg_dist_placement
+ * we continue without waiting.
+ *
  * Before doing any of this it will take an exclusive PlacementCleanup lock.
  * This is to ensure that this function is not being run concurrently.
  * Otherwise really bad race conditions are possible, such as removing all
- * placements of a shard. waitForCleanupLock indicates if this function should
+ * placements of a shard. waitForLocks indicates if this function should
  * wait for this lock or error out.
  *
  */
 int
-DropMarkedShards(bool waitForCleanupLock)
+DropMarkedShards(bool waitForLocks)
 {
 	int removedShardCount = 0;
 	ListCell *shardPlacementCell = NULL;
+	Oid distPlacementId = InvalidOid;
 
 	if (!IsCoordinator())
 	{
 		return 0;
 	}
 
-	if (waitForCleanupLock)
+	if (waitForLocks)
 	{
 		LockPlacementCleanup();
 	}
-	else if (!TryLockPlacementCleanup())
+	else
 	{
-		ereport(WARNING, (errmsg("could not acquire lock to cleanup placements")));
-		return 0;
+		distPlacementId = DistPlacementRelationId();
+		if (!ConditionalLockRelationOid(distPlacementId, AccessShareLock))
+		{
+			ereport(WARNING, (errmsg(
+								  "could not acquire shard lock to cleanup placements")));
```

Comment:
WARNING seems too aggressive

Reply:
It's the same level as the message that's shown if `TryLockPlacementCleanup` fails. Should that be lowered too then?